### DISCUSSION TITLE
automatic connection greyed out

### DISCUSSION THREAD

**Discussion by Renegat27** [blueman-project/blueman#2471] [Q&A] [answered]
blueman: blueman 2.4.2+mint1+wilma
BlueZ: 5.72-0ubuntu5
Distribution: Mint 22
Desktop environment: Cinnamon

* [x ] I have consulted the [Troubleshooting page](https://github.com/blueman-project/blueman/wiki/Troubleshooting) and done my best effort to follow.

**Problem:**
The button for automatic connection on a paired speaker is greyed out, therefore it requires always manual intervention on every startup to connect the bluetooth device.

![Bildschirmfoto vom 2024-08-21 17-20-04](https://github.com/user-attachments/assets/1757f06b-1a83-4bed-8de6-0781542db821)

**Steps taken so far:**
- Verified that AutoEnable=true is configured in /etc/bluetooth/main.conf
- Removed and reconnected the bluetooth device
- Revoked and reapplied trust status
-  Configured/Experimented with ScanIntervalAutoConnect, ScanWindowAutoConnect andAutoconnecttimeout in /etc/bluetooth/main.conf (was a workaround proposed in a Mint forum)
- `blueman-applet --loglevel debug` gave out a couple warnings but no errors

**Steps i came up with but could not do:**
- Test with another bluetooth speaker (I don't have another such device around)
- Test with another adapter (same reason as above)

**Comment by pasha-pivo** [comment]
Same thing. I have a number of devices pared with my laptop. They don't want to be connected to the laptop automatically. On all the devices the second line in the context menu i greyed out and not active. 

There is a config file. Changing parameters in it does not make any difference.

Screenshots:
![image](https://github.com/user-attachments/assets/bce370f8-4771-4ec2-b3bf-08940d198e53)
![image](https://github.com/user-attachments/assets/1b0b4da4-bea4-45c8-a0f5-de3f424f2b63)
![image](https://github.com/user-attachments/assets/d19ef0c8-9fa7-4700-a892-eea74ab87df8)

**Comment by infirit** [comment]
If you are referring to `Automatich verbinden` then that's just a header for the checkbox below it.

**Comment by Renegat27** [comment]
okay, what is the expected behavior when the checkbox is checked?

**Reply by infirit** [reply]
While blueman-applet is running it will try to connect to the device (service really) every 60 seconds.

**Comment by infirit** [comment, marked as answer]
Unfortunately it looks like this broke at some point. Below patch with debug logging should fix this. Can you check if it does for you @Renegat27 . If it does I'll submit a PR for review.

```patch
diff --git a/blueman/plugins/applet/AutoConnect.py b/blueman/plugins/applet/AutoConnect.py
index efb4d16d6..102737282 100644
--- a/blueman/plugins/applet/AutoConnect.py
+++ b/blueman/plugins/applet/AutoConnect.py
@@ -1,10 +1,12 @@
 from gettext import gettext as _
 from typing import TYPE_CHECKING, Union, Optional, Any
+import logging
 
 from gi.repository import GLib
 
 from blueman.Sdp import ServiceUUID
 from blueman.bluez.Device import Device
+from blueman.bluez.errors import BluezDBusException
 from blueman.gui.Notification import Notification
 from blueman.plugins.AppletPlugin import AppletPlugin
 
@@ -40,18 +42,26 @@ class AutoConnect(AppletPlugin):
             self._run()
 
     def _run(self) -> bool:
-        for address, uuid in self.get_option('services'):
-            device = self.parent.Manager.find_device(address)
-            if device is None or device.get("Connected"):
+        logging.debug("Autoconnect run")
+        for object_path, uuid in self.get_option('services'):
+            device = Device(obj_path=object_path)
+            try:
+                address = device["Address"]
+            except BluezDBusException:
+                address = None
+
+            if address is None or device.get("Connected"):
                 continue
 
             def reply(dev: Optional[Device] = device, service_name: str = ServiceUUID(uuid).name) -> None:
                 assert isinstance(dev, Device)  # https://github.com/python/mypy/issues/2608
+                logging.debug(f"Autoconnect {dev.display_name}")
                 Notification(_("Connected"), _("Automatically connected to %(service)s on %(device)s") %
                              {"service": service_name, "device": dev.display_name},
                              icon_name=dev["Icon"]).show()
 
             def err(_reason: Union[Exception, str]) -> None:
+                logging.debug("Autoconnect failed")
                 pass
 
             self.parent.Plugins.DBusService.connect_service(device.get_object_path(), uuid, reply, err)
```

**Reply by Renegat27** [reply]
I can confirm after applying the changes provided by @infirit that it connects now automatically. (instantaneous when opening the application and with a 2-3 min delay on PC startup)

**Reply by cschramm** [reply]
https://github.com/blueman-project/blueman/pull/2103/commits/9d9af921e5c1ae965512a2a4e32daf057536ff38 Broken in all 2.4 versions. :see_no_evil:
